sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -211,6 +217,7 @@ class StreamSuite extends StreamTest {
       CheckSinkLatestBatchId(1),
       // Add some data in batch 2
       AddData(inputData, 7, 8, 9),
+      EnsureManualClockInWaitingState,
```

Comment:
> Is it ever correct to AdvanceManualClock without EnsureManualClockInWaitingState?

It depends. In this case, no. However, there are some cases not need to use `EnsureManualClockInWaitingState` considering ManualClock exists so long without `isWaiting` added in this PR.